### PR TITLE
Add scraping for Java profiles

### LINKED ISSUE #574: configurable profiling config
currently the profile types are hard coded at https://github.com/grafana/k8s-monitoring-helm/blob/335cb5d39f346c4dd18b8649e6c8551016af3049/charts/k8s-monitoring/templates/alloy_config/_profiles_pprof.alloy.txt#L53

as the result `godeltaprof` cannot be configured as https://grafana.com/docs/pyroscope/latest/configure-client/grafana-agent/go_pull/#prepare-the-collector-configuration-file has suggested

### LINKED ISSUE #587: Add the ability to scrape Java Profiles

### PR DESCRIPTION
**Pull request by petewall**
Fixes #574 
Fixes #587 

Adds a section for Java profiles.

### CODE REVIEW

File: charts/k8s-monitoring/templates/alloy_config/_profiles_java.alloy.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  role = "pod"
+}
+
+discovery.relabel "java_pods" {
+  targets = discovery.kubernetes.java_pods.targets
+  rule {
+    action = "drop"
+    regex = "Succeeded|Failed|Completed"
+    source_labels = ["__meta_kubernetes_pod_phase"]
+  }
```
Comment (simonswine) [inline charts/k8s-monitoring/templates/alloy_config/_profiles_java.alloy.txt:23]: In order to be able to get the command line executable you need to also use discovery.process with join. This is how it should look like:

```suggestion
discovery.process "java_pods" {
  join = discovery.kubernetes.java_pods.targets
}

discovery.relabel "java_pods" {
  targets = discovery.process.java_pods.targets
  rule {
    action = "drop"
    regex = "Succeeded|Failed|Completed"
    source_labels = ["__meta_kubernetes_pod_phase"]
  }
  // Filter only java processes
 rule {
        source_labels = ["__meta_process_exe"]
        action = "keep"
        regex = ".*/java$"
  }
```

File: charts/k8s-monitoring/values.yaml [hunk 2]
```diff
@@ -1508,27 +1508,45 @@ profiles:
...
-  # java:
-  #  enabled: true
-  #  namespaces: []
-  #  # -- Rule blocks to be added to the discovery.relabel component for Java profile sources.
-  # ([docs](https://grafana.com/docs/alloy/latest/reference/components/discovery.relabel/#rule-block))
-  #  extraRelabelingRules: ""
+  java:
+    # -- Gather profiles by scraping java HTTP endpoints
+    # @section -- Profiles (java)
+    enabled: true
```
Comment (simonswine) [inline charts/k8s-monitoring/values.yaml:1514]: Just to make sure I am getting this right: Profile collecting is by default off, but once `profiles.enabled: true` the java scraping will be enabled.

**Review by simonswine — APPROVED** [review]
Thank you for working on this. I think we need to filter it down to java processes as per my suggestion. Apart from that looking all good to me.

File: charts/k8s-monitoring/values.yaml (hunk 2 above)
Comment (petewall) [reply charts/k8s-monitoring/values.yaml:1514]: Yep! It all gets enabled when the user sets:
```
profiles:
  enabled: true
```

**Pull request merged**